_source/_posts/2022-05-13-express-prisma-and-tailwindcss.md

```diff
@@ -0,0 +1,665 @@
+> `express-session` is required for passport.
+> `tailwindcss` is the CSS framework you will use.  If you've never used tailwindcss before you may wonder why it's a development dependency.  This is because tailwindcss will dynamically build your CSS files from your views and configuration.  More on that later
+> `prisma` is the ORM you are using.  This is a dev dependency because the prisma library handles the migrations, seed data, etc, while the `@prisma/client` package is used in your application at runtime.
+
+At this point, you should initialize tailwindcss and prisma even though you won't configure to use either quite yet.
+
+Start with prisma.  Run the following command.
+
+```console
+npx prisma init
+```
+
+This command will add a new folder named `prisma` to your application. This also adds a file called `.env` to your project with some default configuration. Replace the contents of that file with the code below.
+
+```bash
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB, and CockroachDB (Preview).
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL={yourServersConnectionString}
+SEED_USER_NAME={yourOktaUserName}
+```
+
+In this example I used Microsoft's SQL Server because that's what I had installed at the time of writing this application.  But one of the cool things about Prisma and most ORMs is that they are built to support many databases.  The boilerplate code includes comments about how to configure your application for other database servers.  Feel free to use what you feel most comfortable with and use the appropriate connection string.
+
+The `SEED_USER_NAME` setting is used in your seed data.  You want to make sure that it's the same as the Okta username you sign in to.  This will allow the application to associate your logged-in user with the data that you will seed into your database.  
+
+ Next, update your `package.json` file with the following code.
+
+```JSON
+{
+  "name": "workout-app",
+  "version": "0.0.0",
+  "private": true,
+  "prisma": {
+    "seed": "node prisma/seed.js"
+  },
+  "scripts": {
+    "start": "node ./bin/www"
+  },
+  "dependencies": {
+    "@prisma/client": "^3.13.0",
+    "cookie-parser": "~1.4.4",
+    "debug": "~2.6.9",
+    "dotenv": "^16.0.0",
+    "express": "~4.16.1",
+    "http-errors": "~1.6.3",
+    "morgan": "~1.9.1",
+    "passport": "^0.5.2",
+    "passport-openidconnect": "^0.1.1",
+    "pug": "2.0.0-beta11"
+  },
+  "devDependencies": {
+    "prisma": "^3.13.0",
+    "tailwindcss": "^3.0.24"
+  }
+}
+```
+
+This will add the `prisma.seed` command you will need later.  
```

Comment:
Should it be `prisma seed`?

Reply:
https://github.com/oktadev/okta-blog/pull/1172/commits/4a53d6be0eae0850d06be9cc2bfdd0615cb7d0d8